app/models/api_docs/service.rb

```diff
@@ -17,6 +18,7 @@ class ApiDocs::Service < ApplicationRecord
   validates :name, :system_name, :base_path, :swagger_version, length: { maximum: 255 }
   validates :description, length: { maximum: 65535 }
   validates :body, length: { maximum: 4294967295, allow_blank: true }
+  validate :service_belongs_to_account, if: -> { service_id.present? && account_id.present? }
```

Comment:
Apparently `account_id` be `nil`. Does it mean that an `ApiDocs::Service` that belongs to no account can also belong to any service?

Reply:
@guicassolato When it is created by UI or API, it can never be nil and it is checked in the tests, but in the model is another story... IMO by logic the account should never be nil (the association is `dependent: :destroy` from the account's side), and it should be `belongs_to: :account, required: :true` to never be nil, but before doing that I would need to check that it is always the case 🤔

Reply:
Ok. It's not this PR's problem anyway. If the `ApiDocs::Service` can belong to any service, it's only because it can also belong to no account, and that's out of scope.

Reply:
Well you can use `belongs_to :account, required: true`
For service just check:

```ruby
service_id.present ? account.services.where(service_id: service_id).exists? : true

```

Reply:
For service is already being checked and imo in a simpler way 😄 For account required, I will add that 😉

Reply:
Good for this 👍 : 
```
return if service_id.blank? || !service_id_changed?
account.services.accessible.where(service_id: service_id).exists?
```

But first:
https://github.com/3scale/porta/pull/86
😄

Reply:
done 😉